_posts/2017-08-24-coming-to-terms-with-a-bug.md

```diff
@@ -0,0 +1,182 @@
+sigaltstack(NULL, {ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}) = 0
+mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f5f6ca6d000
+sigaltstack({ss_sp=0x7f5f6ca6d000, ss_flags=0, ss_size=8192}, NULL) = 0
+getcwd("/home/tom/src/papers", 512)     = 21
+open("/home/tom/src/papers/.env", O_RDONLY|O_CLOEXEC) = 3
+ioctl(3, FIOCLEX)                       = 0
+read(3, "PAPERS_AWS_ACCESS_KEY=<REDACTED>"..., 8192) = 228
+getrandom("", 0, GRND_NONBLOCK)         = 0
+getrandom("\x3d\xfd\x3c\x08\x48\x6d\x03\x13", 8, GRND_NONBLOCK) = 8
+getrandom("\x80\x11\x02\x43\x2e\xa6\x6f\x1b", 8, GRND_NONBLOCK) = 8
+read(3, "", 8192)                       = 0
+close(3)                                = 0
+stat("/home/tom/.terminfo", 0x7ffcda2eb7d0) = -1 ENOENT (No such file or directory)
+stat("/etc/terminfo", 0x7ffcda2eb7d0)   = -1 ENOENT (No such file or directory)
+stat("/lib/terminfo", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
+stat("/lib/terminfo/s/screen", {st_mode=S_IFREG|0644, st_size=1587, ...}) = 0
+open("/lib/terminfo/s/screen", O_RDONLY|O_CLOEXEC) = 3
+ioctl(3, FIOCLEX)                       = 0
+read(3, "\32\1*\0+\0\20\0i\1z\2screen|VT 100/ANSI X"..., 8192) = 1587
+close(3)                                = 0
+stat("/home/tom/.terminfo", 0x7ffcda2eb790) = -1 ENOENT (No such file or directory)
+stat("/etc/terminfo", 0x7ffcda2eb790)   = -1 ENOENT (No such file or directory)
+stat("/lib/terminfo", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
+stat("/lib/terminfo/s/screen", {st_mode=S_IFREG|0644, st_size=1587, ...}) = 0
+open("/lib/terminfo/s/screen", O_RDONLY|O_CLOEXEC) = 3
+ioctl(3, FIOCLEX)                       = 0
+read(3, "\32\1*\0+\0\20\0i\1z\2screen|VT 100/ANSI X"..., 8192) = 1587
+close(3)                                = 0
+ioctl(1, TCGETS, {B38400 opost isig icanon echo ...}) = 0
+futex(0x7f5f6c161048, FUTEX_WAKE_PRIVATE, 2147483647) = 0
+mmap(NULL, 2101248, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7f5f6abff000
+mprotect(0x7f5f6abff000, 4096, PROT_NONE) = 0
+clone(child_stack=0x7f5f6adfee30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tidptr=0x7f5f6adff9d0, tls=0x7f5f6adff700, child_tidptr=0x7f5f6adff9d0) = 13361
+epoll_create1(EPOLL_CLOEXEC)            = 3
+pipe2([4, 5], O_NONBLOCK|O_CLOEXEC)     = 0
+epoll_ctl(3, EPOLL_CTL_ADD, 4, {EPOLLIN|EPOLLET, {u32=4294967295, u64=18446744073709551615}}) = 0
+socket(AF_INET, SOCK_STREAM|SOCK_CLOEXEC, IPPROTO_IP) = 6
+setsockopt(6, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
+```
+
+It looks scary at first, but you absolutely don't need to know what every
+system call does to gain insight from it (if you wonder, they all have manpages
+so you can go `man futex`).
+
+You can follow the program linearly, and it's easy to spot things like [the
+moment we call `dotenv` in
+`config.rs`](https://github.com/store2be/pape-rs/blob/master/src/config.rs#L60).
```

Comment:
Should not be master but a specific commit blob. Otherwise it will break in the future